iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/receiver/protocol/pipeconsensus/PipeConsensusReceiver.java

```diff
@@ -1112,8 +1123,38 @@ private static class PipeConsensusTsFileWriter {
     private File writingFile;
     private RandomAccessFile writingFileWriter;
 
-    public PipeConsensusTsFileWriter(int index) {
+    public PipeConsensusTsFileWriter(int index, ConsensusPipeName consensusPipeName) {
       this.index = index;
+      this.consensusPipeName = consensusPipeName;
+    }
+
+    public void setFilePath(String receiverBasePath) throws IOException {
+      this.localWritingDirPath = receiverBasePath + File.separator + index;
+      File tsFileWriterDirectory = new File(this.localWritingDirPath);
+      // Remove exists dir
+      if (tsFileWriterDirectory.exists()) {
```

Comment:
Might this result in frequent directory creation and deletion? If so, it doesn't seem necessary?

Reply:
@OneSizeFitsQuorum  It will not cause frequent file creation and deletion. This code will only be called when IoTDB is started to ensure that the dir is clean every time IoTDB restarts.